gateways/server/pulsar/start.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2019 Lucidworks Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pulsar
+
+import (
+	"github.com/apache/pulsar-client-go/pulsar"
+	"github.com/argoproj/argo-events/common"
+	"github.com/argoproj/argo-events/gateways"
+	"github.com/argoproj/argo-events/gateways/server"
+	"github.com/argoproj/argo-events/pkg/apis/eventsources/v1alpha1"
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+	"k8s.io/client-go/kubernetes"
+	"time"
+)
+
+// EventListener implements Eventing for Pulsar event source
+type EventListener struct {
+	Logger *logrus.Logger
+	// k8sClient is kubernetes client
+	K8sClient kubernetes.Interface
+}
+
+// StartEventSource starts an event source
+func (listener *EventListener) StartEventSource(eventSource *gateways.EventSource, eventStream gateways.Eventing_StartEventSourceServer) error {
```

Comment:
This approach as previously implemented had a memory leak in certain edge scenarios. Can you update the implementation to look something similar to https://github.com/argoproj/argo-events/blob/master/gateways/server/amqp/start.go#L40 ?